promql/engine.go

```diff
@@ -481,12 +481,19 @@ func (ng *Engine) populateSeries(ctx context.Context, q storage.Queryable, s *Ev
 	Inspect(s.Expr, func(node Node, path []Node) error {
 		var set storage.SeriesSet
 		params := &storage.SelectParams{
-			Step: int64(s.Interval / time.Millisecond),
+			Start: timestamp.FromTime(s.Start.Add(-LookbackDelta)),
```

Comment:
With apologies for being late to the party, I only noticed this change when I was fixing some merge conflicts: why are we now loading `LookbackDelta` worth of data before range start for matrix selectors?

I was doing the same in my (thus far) ill fated `xrate` attempt and there was a noticeable performance penalty to loading an extra 5 minutes of data for every single matrix query when only `xrate` needed it. IIRC, it was around 10%, which is non-negligible. AFAIK no existing function actually uses any samples outside of the range interval.

Reply:
You're right, of course. My changes were to the `Inspect` call above, where `mint` is calculated, not this one. Probably worth fixing, though.

Reply:
Yes, thanks for catching this. Would you like to send a PR?